pytorch_lightning/trainer/training_loop.py

```diff
@@ -751,3 +744,23 @@ def training_forward(self, batch, batch_idx, opt_idx, hiddens):
         output = self.process_output(output, train=True)
 
         return output
+
+    def update_learning_rates(self, interval):
+        ''' Update lr '''
+        if self.lr_schedulers is not None:
+            for lr_scheduler in self.lr_schedulers:
+                current_step_idx = self.batch_idx if interval == 'batch' else self.current_epoch
+                # Take step if call to update_learning_rates matches the interval key and
+                # the current step modulo the schedulers frequency is zero
+                if lr_scheduler['interval'] == interval and current_step_idx % lr_scheduler['frequency'] == 0:
```

Comment:
well what if the `interval` is "epoch" and here you compare to `current_step_idx`

Reply:
`current_step_idx` is set differently, depending if it is called after a training step or after an epoch
```
current_step_idx = self.batch_idx if interval == 'step' else self.current_epoch
```
name is maybe a bit confusing. Will change to just `current_idx`

Reply:
`current_idx` would be better :]